test/Microsoft.AspNet.Mvc.Formatters.Json.Test/JsonInputFormatterTest.cs

```diff
@@ -150,6 +158,61 @@ public static IEnumerable<object[]> JsonFormatterReadSimpleTypesData
         }
 
         [Fact]
+        public async Task ReadAsync_InvalidArray_AddsOverflowErrorsToModelState()
+        {
+            // Arrange
+            var content = "[0, 23, 300]";
+            var formatter = new JsonInputFormatter();
+            var contentBytes = Encoding.UTF8.GetBytes(content);
+
+            var modelState = new ModelStateDictionary();
+            var httpContext = GetHttpContext(contentBytes);
+            var provider = new EmptyModelMetadataProvider();
+            var metadata = provider.GetMetadataForType(typeof(byte[]));
+            var context = new InputFormatterContext(
+                httpContext,
+                modelName: string.Empty,
+                modelState: modelState,
+                metadata: metadata);
+
+            // Act
+            var result = await formatter.ReadAsync(context);
+
+            // Assert
+            Assert.True(result.HasError);
+            Assert.Equal("The supplied value is invalid for Byte.", modelState["[2]"].Errors[0].ErrorMessage);
+            Assert.Null(modelState["[2]"].Errors[0].Exception);
+        }
+
+        [Fact]
+        public async Task ReadAsync_InvalidComplexArray_AddsOverflowErrorsToModelState()
+        {
+            // Arrange
+            var content = "[{name: 'Name One', Age: 30}, {name: 'Name Two', Small: 300}]";
```

Comment:
Why add that redundancy? Functional tests are about mainline scenarios and not uncommon binding errors.